lib/gst/player/gstplayer.c

```diff
@@ -1226,6 +1227,36 @@ request_state_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
 }
 
 static void
+tags_cb (GstBus * bus, GstMessage *msg, gpointer user_data)
+{
+  GstPlayer *self = GST_PLAYER (user_data);
+  GstTagList *tags = NULL;
+
+  gst_message_parse_tag (msg, &tags);
+
+  if (gst_tag_list_get_scope (tags) == GST_TAG_SCOPE_GLOBAL) {
+    g_mutex_lock (&self->lock);
+    if (self->media_info) {
+      if (self->media_info->tags) {
+        gst_tag_list_unref (self->media_info->tags);
+        self->media_info->tags = tags;
+      }
+    }
+    else {
+      if (self->global_tags)
+        gst_tag_list_unref (self->global_tags);
+      self->global_tags = tags;
+    }
+    g_mutex_unlock (&self->lock);
+
+    GST_DEBUG_OBJECT (self, "updating global tags");
```

Comment:
You probably want to emit the media-info-updated signal here as the tags have updated (but only if we had a media-info)